core/test/utils/utils_test.cpp

```diff
@@ -0,0 +1,236 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <complex>
+#include <tuple>
+#include <type_traits>
+
+
+#include <gtest/gtest.h>
+
+
+#include "core/test/utils.hpp"
+
+
+namespace {
+
+
+using i_type = std::integral_constant<int, 42>;
+using t_type = std::tuple<int>;
+
+using testing_types1 = testing::Types<double>;
+using testing_types2 = testing::Types<t_type, int>;
+using testing_types3 = testing::Types<i_type, short, float>;
+
+using tuple_types1 = std::tuple<double>;
+using tuple_types2 = std::tuple<t_type, int>;
+using tuple_types3 = std::tuple<i_type, short, float>;
+
+template <typename... Args>
+struct type_list {};
+
+
+TEST(TypeListHelper, ChangeOuterWrapper1)
+{
+    testing::StaticAssertTypeEq<
+        gko::test::change_outer_wrapper_t<std::tuple, testing_types1>,
+        tuple_types1>();
+    testing::StaticAssertTypeEq<
+        gko::test::change_outer_wrapper_t<std::tuple, testing_types2>,
+        tuple_types2>();
+    testing::StaticAssertTypeEq<
+        gko::test::change_outer_wrapper_t<std::tuple, testing_types3>,
+        tuple_types3>();
```

Comment:
Does this test give any additional information than the two before? (same for the one below)

Reply:
But what is the difference to the two tests before? The first test uses a list of length 1 and the second uses a list of length two. I think a list with one element and a list with two elements can be considered to be quite different, but I don't think there is any more difference between a list of two elements and one with three elements.

Reply:
That is true. The handling of two or three elements should be the same. I mainly added it for 2 purposes:
1. Make sure `testing_typesX` has the same contents as `tuple_typesX`. This should prevent the test types from being modified in the wrong way.
2. Make sure the parameter packs work as intended and are not just special cases for 1-2 parameters.
The 2nd point could also be checked by looking at the implementation, but I thought an additional test wouldn't hurt.

If you think I should remove the 3-element tests, I can do so.

Reply:
Well, I don't care that much about it.